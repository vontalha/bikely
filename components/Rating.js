import React, {useState} from 'react';
import {StyleSheet, Text, View, Image, TouchableOpacity} from 'react-native';
import emptyStar from '../assets/emptyStar.png';
import filledStar from '../assets/filledStar.png';

const StarRating = () => {
  const [rating, setRating] = useState(0);

  const handleRatingPress = selectedRating => {
    setRating(selectedRating);
  };

  const renderStars = () => {
    const stars = [];
    const totalStars = 5;

    for (let i = 1; i <= totalStars; i++) {
      const starIcon = i <= rating ? filledStar : emptyStar;

      stars.push(
        <TouchableOpacity
          key={i}
          activeOpacity={0.7}
          onPress={() => handleRatingPress(i)}>
          <Image source={starIcon} style={styles.star} />
        </TouchableOpacity>,
      );
    }

    return stars;
  };

  return (
    <View style={styles.starRatingContainer}>
      <Text style={styles.ratingText}>Rating:</Text>
      <View style={styles.starsContainer}>{renderStars()}</View>
    </View>
  );
};

const styles = StyleSheet.create({
  starRatingContainer: {
    alignItems: 'center',
    marginTop: 20,
  },
  ratingText: {
    fontSize: 18,
    fontWeight: '600',
    marginBottom: 10,
  },
  starsContainer: {
    flexDirection: 'row',
  },
  star: {
    width: 30,
    height: 30,
    marginHorizontal: 2,
  },
});

export default StarRating;
