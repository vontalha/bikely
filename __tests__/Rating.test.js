import React from 'react';
import renderer from 'react-test-renderer';
import Rating from '../components/Rating';

describe('Rating', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Rating />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
