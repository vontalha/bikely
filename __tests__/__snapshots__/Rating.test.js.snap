// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "600",
        "marginBottom": 10,
      }
    }
  >
    Rating:
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/emptyStar.png",
          }
        }
        style={
          {
            "height": 30,
            "marginHorizontal": 2,
            "width": 30,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/emptyStar.png",
          }
        }
        style={
          {
            "height": 30,
            "marginHorizontal": 2,
            "width": 30,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/emptyStar.png",
          }
        }
        style={
          {
            "height": 30,
            "marginHorizontal": 2,
            "width": 30,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/emptyStar.png",
          }
        }
        style={
          {
            "height": 30,
            "marginHorizontal": 2,
            "width": 30,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/emptyStar.png",
          }
        }
        style={
          {
            "height": 30,
            "marginHorizontal": 2,
            "width": 30,
          }
        }
      />
    </View>
  </View>
</View>
`;
